api/src/main/java/org/openmrs/module/rwandahivflowsheet/impl/pih/ProphylaxisMapping.java

```diff
@@ -111,14 +112,14 @@ public boolean getProphylaxisOther() {
 		public String getProphylaxisTypeOther(){
 			if (drugOrder != null && drugOrder.getConcept() != null 
 					&& !isCotrimoxisole() && !isFluconazole() && !isDapsone())
-				return drugOrder.getConcept().getBestName(Context.getLocale()).getName();
+				return drugOrder.getConcept().getName(Context.getLocale()).getName();
 			return "";
 		}
 		
 		public Date getStopDate(){
 			if (drugOrder != null){
-				if (drugOrder.getDiscontinuedDate() != null)
-					return drugOrder.getDiscontinuedDate();
+				if (drugOrder.getEffectiveStopDate() != null)
+					return drugOrder.getEffectiveStopDate();
 				if (drugOrder.getAutoExpireDate() != null)
```

Comment:
By using "getEffectiveStopDate", this method already returns either the dateStopped or the autoExpireDate, whichever is appropriate.  So you can get rid of this second check against drugOrder.getAutoExpireDate and just check the getEffectiveStopDate

Reply:
[Done](https://github.com/PIH/openmrs-module-rwandahivflowsheet/pull/2/commits/9478de14df21fce4aebb0e1cacfad3baf8b0c061)